File: server/src/main/java/fr/pacpilot/server/platform/PacPilotApplication.java
package fr.pacpilot.server.platform;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

/**
 * Entry point of the modular monolith.
 *
 * <p>Lives in {@code platform} — the cross-cutting package. Component scanning starts here and
 * reaches every bounded context under {@code fr.pacpilot.server}, but contexts must not reach into
 * one another: that wall is enforced by ArchUnit at M4.
 */
@SpringBootApplication(scanBasePackages = "fr.pacpilot.server")
public class PacPilotApplication {

    public static void main(String[] args) {
        SpringApplication.run(PacPilotApplication.class, args);
    }
}

File: server/src/main/kotlin/fr/pacpilot/server/platform/PacPilotApplication.kt
package fr.pacpilot.server.platform

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication

/**
 * Entry point of the modular monolith.
 *
 * Lives in `platform` — the cross-cutting package. Component scanning starts here and reaches every
 * bounded context under `fr.pacpilot.server`, but contexts must not reach into one another:
 * that wall is enforced by ArchUnit at M4.
 */
@SpringBootApplication(scanBasePackages = ["fr.pacpilot.server"])
class PacPilotApplication

fun main(args: Array<String>) {
    runApplication<PacPilotApplication>(*args)
}
